test/on.js

```diff
@@ -0,0 +1,15 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+
+describe('on', function() {
+  it('compares identity using identity', function() {
+    const eqBy = R.on((a, b) => a === b);
+    eq(eqBy(R.identity, R.identity, R.identity), true);
+  });
```

Comment:
I would drop this test.  I find it confusing, and it's hard to see what `on` is doing here.  And of course `eqBy(identity)` is already covered with `identical`.